sn_networking/src/lib.rs

```diff
@@ -633,16 +636,55 @@ impl Network {
     }
 
     /// Put `Record` to network
-    pub async fn put_record(&self, record: Record) -> Result<()> {
+    /// optionally verify the record is stored after putting it to network
+    pub async fn put_record(&self, record: Record, verify_store: bool) -> Result<()> {
+        let the_record = record.clone();
         debug!(
             "Putting record of {:?} - length {:?} to network",
-            record.key,
+            the_record.key,
             record.value.len()
         );
         // Waiting for a response to avoid flushing to network too quick that causing choke
         let (sender, receiver) = oneshot::channel();
         self.send_swarm_cmd(SwarmCmd::PutRecord { record, sender })?;
-        receiver.await?
+        let response = receiver.await?;
+
+        if !verify_store {
+            return response;
+        }
+
+        // Verify the record is stored
+        let mut verification_attempts = 0;
+        let mut something_was_found = false;
+        while verification_attempts < VERIFICATION_ATTEMPTS {
+            match self.get_record_from_network(the_record.key.clone()).await {
+                Ok(returned_record) => {
+                    // if the returned register is not _the same_ lets inform the user
+                    if the_record != returned_record {
+                        something_was_found = true;
```

Comment:
the var `something_was_found` shall be reset to `false` at the beginning of each iteration.